src/main/java/com/cloudfiveapp/push/CloudFivePush.java

```diff
@@ -308,7 +338,47 @@ private void notifyCloudFive() {
         }
     }
 
+    private void unregisterCloudFive() {
+        HttpClient httpclient = new DefaultHttpClient();
+        Log.i("CloudFivePush", "unregistering for push notification with registrationId: " + registrationId + " and userIdentifier: " + userIdentifier);
+        HttpPost httppost = new HttpPost("https://www.cloudfiveapp.com/push/unregister");
+
+        try {
+            // Add your data
+            List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(2);
+            //:device_token, :device_model, :device_name, :device_version, :app_version
+            nameValuePairs.add(new BasicNameValuePair("device_token", registrationId));
+            nameValuePairs.add(new BasicNameValuePair("package_name", applicationContext.getPackageName() ));
+            nameValuePairs.add(new BasicNameValuePair("device_model", android.os.Build.MODEL));
+            nameValuePairs.add(new BasicNameValuePair("device_name", android.os.Build.DISPLAY));
+            nameValuePairs.add(new BasicNameValuePair("device_version", android.os.Build.VERSION.RELEASE));
+            nameValuePairs.add(new BasicNameValuePair("device_identifier",
+                    Settings.Secure.getString(applicationContext.getContentResolver(), Settings.Secure.ANDROID_ID)));
+            nameValuePairs.add(new BasicNameValuePair("device_platform", "android"));
+            if (userIdentifier != null) {
+                nameValuePairs.add(new BasicNameValuePair("user_identifier", userIdentifier));
+            }
+
+            String version;
+            try {
+                version = applicationContext.getPackageManager().getPackageInfo(applicationContext.getPackageName(), 0).versionName;
+            } catch (PackageManager.NameNotFoundException e) {
+                version = "unknown";
+            }
+            nameValuePairs.add(new BasicNameValuePair("app_version", version));
+            httppost.setEntity(new UrlEncodedFormEntity(nameValuePairs));
+            HttpResponse response = httpclient.execute(httppost);
+        } catch (ClientProtocolException e) {
+            // TODO Auto-generated catch block
+        } catch (IOException e) {
```

Comment:
You should combine these catch blocks so the logging still happens `catch (ClientProtocolException | IOException e) {`